### PR TITLE
Update `Dropdown` code documentation

### PR DESCRIPTION
**Pull request by alex-ju**
### :pushpin: Summary

Updating the technical documentation ('Code' tab) for `Dropdown` component to bring it up to date with the newly released changes.

### :hammer_and_wrench: Detailed description

Aside from the API updates that are part of this PR, we would likely want to showcase the new `listItems` (Checkmark, Checkbox, and Radio), but before doing so I'd like to check with design and see if they're better fit and more relevant under the upcoming Filter pattern.

### :link: External links

Jira ticket: [HDS-1702](https://hashicorp.atlassian.net/browse/HDS-1702)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-1702]: https://hashicorp.atlassian.net/browse/HDS-1702?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/docs/components/dropdown/partials/code/component-api.md [hunk 1]
```diff
@@ -41,6 +59,18 @@ The Dropdown component is composed of different child components each with their
   </C.Property>
   <C.Property @name="color" @type="enum" @values={{array "primary" "secondary" }} @default="primary"/>
   <C.Property @name="size" @type="enum" @values={{array "medium" "small" }} @default="medium"/>
+  <C.Property @name="icon" @type="string">
+    Acceptable value: any [icon](/icons/library) name.
```
Comment (MelSumner) [inline website/docs/components/dropdown/partials/code/component-api.md:63]:
```suggestion
    Used to show an icon. Any [icon](/icons/library) name is accepted.
```

Suggestion based on prior art: https://github.com/hashicorp/design-system/pull/1261/files and looking at other component docs files. 

But there are a lot of components that we may want to go through and align on this, so it's a non-blocking suggestion.

File: website/docs/components/dropdown/partials/code/component-api.md [hunk 2]
```diff
@@ -137,6 +167,98 @@ The Dropdown component is composed of different child components each with their
...
+
+<Doc::ComponentApi as |C|>
+  <C.Property @name="yield">
+    Content to be used in the item.
+  </C.Property>
+  <C.Property @name="selected" @type="boolean" @values={{array "false" "true" }} @default="false">
+    Displays a checkmark symbol indicating the current selection.
+  </C.Property>
+  <C.Property @name="icon" @type="string">
+    Acceptable value: any [icon](/icons/library) name.
```
Comment (MelSumner) [inline website/docs/components/dropdown/partials/code/component-api.md:180]: Same as above; suggestion, non-blocking

**Review by MelSumner — APPROVED** [review]
Made a couple of non-blocking suggestions. Otherwise looks good. 👍

**Comment by didoo** [comment]
> In the interest of keeping the docs in sync with the code changes I'm going to bring these in

@Dhaulagiri I didn't review the ticket (at all). What do we do? Review and open a Jira ticket with things that we may want to fix?

**Pull request merged**